packages/amplify-provider-awscloudformation/lib/transform-graphql-schema.js

```diff
@@ -22,11 +26,61 @@ const schemaDirName = 'schema';
 
 function failOnInvalidAuthType(usedDirectives, opts) {
   if (usedDirectives.includes('auth') && !opts.isUserPoolEnabled) {
-    throw new Error(`You are trying to use the @auth directive without enabling Amazon Cognito user pools for your API.
-Run \`amplify update api\` and choose "Amazon Cognito User Pool" as the authorization type for the API.`);
+    throw new Error(`You are trying to
+use the @auth directive without enabling Amazon Cognito user pools for your API.
+Run \`amplify update api\` and choose
+"Amazon Cognito User Pool" as the authorization type for the API.`);
   }
 }
 
+function warnOnAuth(context, map) {
+  const unAuthModelTypes = Object.keys(map).filter(type => (!map[type].includes('auth') && map[type].includes('model')));
+  if (unAuthModelTypes.length) {
+    context.print.warning('\nThe following types do not have \'@auth\' enabled.Consider using @auth with @model');
+    context.print.warning(unAuthModelTypes.map(type => `\t - ${type}`).join('\n'));
+    context.print.info('Learn more about @auth here: https://aws-amplify.github.io/docs/cli-toolchain/graphql#auth');
+  }
+}
+
+/**
+ * @TODO Include a map of versions to keep track
+ */
+async function transformerVersionCheck(
+  context, resourceDir, cloudBackendDirectory,
+  updatedResources, usedDirectives,
+) {
+  const versionChangeMessage = 'The default behaviour for @auth has changed in the latest version of Amplify\nRead here for details: https://aws-amplify.github.io/docs/cli-toolchain/graphql#authorizing-subscriptions';
+  let transformerConfig;
+  // this is where we check if there is a prev version of the transformer being used
+  // by using the transformer.conf.json file
+  try {
+    transformerConfig = await readTransformerConfiguration(cloudBackendDirectory);
+  } catch (err) {
+    // check local resource if the question has been answered before
+    transformerConfig = await readTransformerConfiguration(resourceDir);
+  }
+  const resources = updatedResources.filter(resource => resource.service === 'AppSync');
+  if (!transformerConfig.Version && usedDirectives.includes('auth')
+  && resources.length > 0) {
+    if (context.exeInfo &&
+      context.exeInfo.inputParams && context.exeInfo.inputParams.yes) {
+      context.print.warning(`\n${versionChangeMessage}\n`);
+    } else {
+      const response = await inquirer.prompt({
+        name: 'transformerConfig',
+        type: 'confirm',
+        message: `${versionChangeMessage} \n Do you wish to continue?`,
```

Comment:
We're printing this message anyway, but in one case as part of a prompt, otherwise as a warning, why not print it as warning anyway before the if, then have the prompt if needed?

nit: I'd not add newlines to messages as prefix/suffix, since once we'll add logging it will be hard to cleanup all these 'additions'.